Add information about sizing issues (#1943)
sourcegraph/handbook: content/departments/product-engineering/engineering/code-graph/code-insights/processes.md

  - Engineers, designer, PM and EM **_propose_ issues to be worked on** the week _before_ the mid-iteration [Monday sync](#weekly-sync) by adding them to the next iteration milestone and setting "Status" to "Proposed". This is done considering our [higher-level goals and roadmap](../../../../../strategy-goals/strategy/code-graph/code-insights/index.md).

>     - The issue should also have its **_Size_ column filled out**, so that it can be evaluated whether it fits into the iteration. If the proposer lacks the information to estimate the issue, they reply on the issue in GitHub or raise it in our Slack channel to get the missing information or get an estimate from the appropiate person. Teammates may also discuss this in ad-hoc synchronous meetings if beneficial. An assignee may also already volunteer or be proposed, but this may still be changed at the [Monday sync](#weekly-sync) to distribute workload. See below for information on issue sizing.

Reviewer: I know you didn't add this, but as I read this I find it a little confusing. Why would someone be proposing an issue if they lack the context on the issue? In the case of a PM or EM, that may make sense, but in general I'd expect someone who doesn't have the context to not be able to advocate for an issue.

Author: I think this is fine personally. I could see a description sounding like something you want to see happen without knowing the implementation details enough to size it. This point here helps to point out that you should got _get_ context before trying to propose an issue to be worked on.

- If one or more issues that were planned for an iteration are looking to **not get finished** (which includes testing) in the [current iteration](https://github.com/orgs/sourcegraph/projects/200/views/1) (while maintaining sustainable work practices) the assignee **raises this as soon as possible asynchronously** to the team (including the PM and EM), e.g. on the GitHub issue or Slack. These issues then **become _proposed_ issues for the next iteration** (meaning nothing carries over automatically, but we also don't just drop and forget missed issues).

## How we think about issue sizes

> When sizing an issue please keep in mind the size includes the entire lifecycle of the issue. The coding, any testing both automated and manual, as well as PR reviews, documentation, and any follow-up needed.

Reviewer: > and any follow-up needed

This sticks out to me as being fairly well outside the bounds of a single issue. If there is truly some follow up tasks, should that not be separate issues? If there is unknown follow up, how would we plan to include that?

Author: Excellent point! Re-wording.

> When thinking about the size of an issue we try to stay away from actual time estimates. But generally, 1 and 2 sizes would probably finish in less than a full workday. 3 and 5 probably no more than 2 workdays. 8 and 13 get closer to a full week and beyond that begins to span whole weeks or even months.

Reviewer: I'm a little uncomfortable with this, it feels very contradictory. If I were reading this for the first time, I'm not sure what to think - the first line says not to think in terms of time, but the only frame of reference I'm being given is time.

It might be useful to instead reframe in terms of something else?
```suggestion
When thinking about the size of an issue we try to stay away from actual time estimates. Generally, 1 and 2 sizes should be considered small. 3 and 5 are medium issues. 8 and 13 can be considered large issues. Any number larger than 13 can be considered significantly large, and possibly could represent a small project.
```

While there is no "maximum" number, once we get higher than a 13 in size it will probably prompt some discussion or at least need some explanation in the issue.

> ### Examples

Reviewer: Something that sticks out to me is that all of these examples are framed around a well-defined problem. For example, "add x feature to y page" might be a 13 point work effort, but it still is a well-defined problem.

There is an entirely separate category of issue that is "thing is broken, and we don't know why". Do you have any thoughts on how we can provide examples for these cases?

Author: This sounds like a spike to me. Anything with that size of an unknown needs discovery before it can be estimated. I'll think on this for a bit and see if I can get a section started on spikes.

> - 1: An issue that you completely know the entire solution and are 100% confident in how long it will take you. You have no questions and no dependencies on other teammates. There aren't any tests that need writing. Fixing a typo, updating copy, or very minor CSS changes would most likely all be 1s.

Reviewer: Since the above section specifies 
> we try to stay away from actual time estimates

It might make sense to reframe this point around scope, rather than how long it will take.

```suggestion
- 1: A small issue that you completely know the entire solution and are 100% confident in the scope. You have no questions and no dependencies on other teammates. There aren't any tests that need writing. Fixing a typo, updating copy, or very minor CSS changes would most likely all be 1s.
```
